library/kotlin/src/io/envoyproxy/envoymobile/filters/FilterDataStatus.kt

```diff
@@ -44,5 +44,15 @@ sealed class FilterDataStatus {
    * This may be called by filters which must parse a larger block of the incoming data before
    * continuing processing, and will handle their own buffering.
    */
-  class StopIterationNoBuffer : FilterDataStatus()
+  class StopIterationNoBuffer<T : Headers> : FilterDataStatus<T>()
+
+  /**
+   * Resume previously-stopped iteration, possibly forwarding headers, if iteration was previously
+   * stopped during an on*Headers invocation.
+   *
+   * It is an error to return ResumeIteration if iteration is not currently stopped, and it is
+   * an error to include headers if headers have already been forwarded to the next filter
+   * (i.e. iteration was stopped during an on*Data invocation instead of on*Headers).
+   */
```

Comment:
I think it's worth including details as to why trailers are not specified here, similarly to how you have in the `StopIteration` case definition

Reply:
I'm a little bit concerned that describing an option that's not present might be more confusing than simply describing those that are.